pmd-core/src/main/java/net/sourceforge/pmd/util/OptionalBool.java

```diff
@@ -6,16 +6,48 @@
 
 /** Represents a boolean that may not be present. Use as a non-null type. */
 public enum OptionalBool {
-    YES, NO, UNKNOWN;
+    NO, UNKNOWN, YES;
 
+    /**
+     * <pre>{@code
+     * yes -> no
+     * unk -> unk
+     * no -> yes
+     * }</pre>
+     */
     public OptionalBool complement() {
         switch (this) {
-        case YES: return NO;
-        case NO: return YES;
-        default: return this;
+        case YES:
+            return NO;
+        case NO:
+            return YES;
+        default:
+            return this;
         }
     }
 
+    public static OptionalBool max(OptionalBool a, OptionalBool b) {
+        return a.compareTo(b) > 0 ? a : b;
+    }
+
+    public static OptionalBool min(OptionalBool a, OptionalBool b) {
+        return a.compareTo(b) < 0 ? a : b;
+    }
+
+    /**
+     * <pre>{@code
+     * yes, yes -> yes
+     * yes, unk -> unk
+     * unk, unk -> unk
+     * no, unk -> unk
+     * no, yes -> unk
+     * no, no -> no
+     * }</pre>
+     */
+    public static OptionalBool mix(OptionalBool a, OptionalBool b) {
```

Comment:
I'm unsure about the naming of this method… maybe `both` could be a better alternative?

Reply:
maybe… I've been looking into 3-value logic standard operations for a better name, but no table I found matches this :(

Reply:
I did the same aha

Reply:
I went for `join`